File: eKucniLjubimci.Services/Database/Prodavac.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eKucniLjubimci.Services.Database
{
    public class Prodavac
    {
        public int ProdavacId { get; set; }
        public string PoslovnaJedinica { get; set; }
        public bool isDeleted { get; set; } = false;
        public string SlikaProdavca { get; set; }


        public int OsobaId { get; set; }
        public virtual Osoba Osoba { get; set; }

        public virtual KorisnickiNalog KorisnickiNalog { get; set; }
        public int KorisnickiNalogId { get; set; }


        public virtual List<Novost> Novosti { get; set; }
    }
}

File: flutterUI/flutter_mobile/lib/models/prodavac.dart
import 'package:json_annotation/json_annotation.dart';

import 'korisnicki_nalog.dart';
import 'osoba.dart';

part 'prodavac.g.dart';

@JsonSerializable()
class Prodavac {
  int? prodavacId;
  String? poslovnaJedinica;
  String? slikaProdavca;
  int? osobaId;
  Osoba? osoba;
  KorisnickiNalog? korisnickiNalog;
  int? korisnickiNalogId;

  Prodavac({
    this.prodavacId,
    this.poslovnaJedinica,
    this.slikaProdavca,
    this.osobaId,
    this.osoba,
    this.korisnickiNalog,
    this.korisnickiNalogId,
  });

  factory Prodavac.fromJson(Map<String, dynamic> json) =>
      _$ProdavacFromJson(json);

  Map<String, dynamic> toJson() => _$ProdavacToJson(this);
}
